Server/src/routes/fileParserRouter.ts

```diff
@@ -9,13 +9,24 @@ interface MulterRequest extends Request {
 }
 
 let upload = multer({ dest: 'upload', limits: { fileSize: 8000000 } });
-let router = Router();
 
 /**
  * This route will accept a file and forward to the router. It is first processed by multer middleware,
- * and the file is stored in a temporary directory called tmp/json. This route is referred for processing by 
+ * and the file is stored in a temporary directory called upload/. This route is referred for processing by 
  * the service.
  */
+let router = Router();
+
+router.post('/api/v1/fileParser', [JWTAuthenticator.verifyJWT, upload.single('file')], (request: MulterRequest, response: Response, next: NextFunction) => {
+  try {
+    let dataExtract = new dataExtractionController(request.file.path, request.file.originalname);
+    dataExtract.createRequest(request, response);
+  } catch (error) {
+    if (!request.file)
+      return response.json('Please select a file to upload');
+  }
+})
+
 router.post('/api/v1/dataExtract', [JWTAuthenticator.verifyJWT, upload.single('file')], (request: MulterRequest, response: Response, next: NextFunction) => {
```

Comment:
i think you can remove the dataExtraction route, but can you also make sure that any frontend code that sends a file to /api/v1/dataExtract sends the file to fileParser instead. the endpoint can accept any file from what I remember

Reply:
You're right, I thought it'd be a huge refactor, but it's good now